File: 26. Remove Duplicates from Sorted Array/solution_1.cpp
class Solution 
{
    public:
        int removeDuplicates(vector<int>& nums) 
        {
            int index = 1;
            for (int i = 0; i < size(nums) - 1; ++i)
            {
                if (nums[i] != nums[i + 1])
                {
                    nums[index++] = nums[i + 1];
                }
            }
            
            return index;
        }
};

File: 28. Find the Index of the First Occurrence in a String/solution_1.cs
public class Solution
{
    public int StrStr(string haystack, string needle)
    {
        if (haystack.Length < needle.Length)
            return -1;

        int index = -1;

        for (int i = 0; i < haystack.Length; ++i)
        {
            bool ok = true;

            for (int j = 0; j < needle.Length; ++j)
            {
                if (haystack[i + j] != needle[j])
                {
                    ok = false;
                    break;
                }
            }

            if (ok)
            {
                index = i;
                break;
            }
        }

        return index;
    }
}